lib/publiccloud/instance.pm

```diff
@@ -352,40 +352,65 @@ sub wait_for_ssh {
     my $start_time = time();
     my $check_port = 1;
     my $sleep_period = $args{ignore_wrong_pubkey} ? 20 : 1;
-
+    #
+    my @sem = (1, 1, 1);
```

Comment:
I didn't fully get why we need this one ?

Reply:
Those are semaphores: like `check_port` for ssh port, each `sem[j]=1`, when condition of check-type j is detected `true`, it marks as **done** the test assuming value `sem[j]=0`, that will no more executed on next loops, and when all 3 condition are _true_, that is $pass==3, the test PASS.

Reply:
But all this structure is to run the 3 check types in each test-run, but after analyzed the last tests, I'd propose to use the only type n.1: 
`sudo systemctl is-system-running --wait`